src/commands/deploy/deploy.ts

```diff
@@ -62,6 +63,35 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
         throw new Error(localize('pythonNotAvailableOnWindows', 'Python projects are not supported on Windows Function Apps. Deploy to a Linux Function App instead.'));
     }
 
+    const showCoreToolsWarningKey: string = 'showCoreToolsWarning';
```

Comment:
I think a `Memento` is a better place for this, i.e. `ext.context.globalState`. This would not sync across machines (by default) (which I think is the _correct_ behavior, different machines likely have different versions), and would also not pollute user `settings.json`.

Reply:
This isn't a reason not to use the `ext.context.globalState` but `showCoreToolsWarning` is setting that already exists that Megan was leveraging. The idea was that if someone had already ignored the warning when their version was out of date, then it would remained ignored.

Also, when using `ext.context.globalState`, it can be difficult for users to "un-Don't Show Again" in the event of a mis-click whereas the setting is pretty easy to discover/delete since it'll show up in SCM.

Reply:
Ah, I didn't realize a setting already existed. I think that's reasonable then.